python/src/iceberg/schema.py

```diff
@@ -16,18 +16,22 @@
 # under the License.
 # pylint: disable=W0511
 
-from __future__ import annotations
```

Comment:
Is this incompatible with Pydantic?

Reply:
There is a bug in Pydantic in combination with Python 3.8. Somewhere in a type reference of NestedField something is off. I want to dig into this, but I first wanted to get this PR ready to get feedback on it.